purchase_order_approved/models/purchase_order.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    state = fields.Selection(selection_add=[("approved", "Approved")])
```

Comment:
I think you can move the state to the proper position using this, needs to be tested.
```suggestion
    state = fields.Selection(selection_add=[("to_approve",), ("approved", "Approved")])
```